### ISSUE TITLE
Random freezes on 5.4.* kernels 

### ISSUE THREAD

**Issue by kvaps** [DisplayLink/evdi#179]
Initially please make sure the problem hasn't been already addressed:
  * [X] Are you using the latest driver?
    (https://www.displaylink.com/downloads/ubuntu)
  * [X] Are you using the latest EVDI version?
    (https://github.com/DisplayLink/evdi/releases)
  * [X] If you are using a DisplayLink device, have you checked 'troubleshooting'
    on DisplayLink's website?
    (https://support.displaylink.com/knowledgebase/topics/103927-troubleshooting-ubuntu)
  * [X] Is this issue related to evdi/kernel?
    (if it is rather connected to DisplayLinkManager please take a look at support
    https://support.displaylink.com or forum https://www.displaylink.org/forum/)

Some information for baseline would help a lot:
  * Linux distribution and its version: Arch Linux (upgrade 2020.01.08)
  * Linux kernel version: linux-5.4.2.arch1-1 and linux-5.4.8.arch1-1
  * Xorg version (if used) 1.20.6
  * Desktop environment in use: Xorg + awesome wm

Hi, after upgrading my distro evdi started freezing quite often, it is usually happens every 1-2 minute with duration about 10-15 seconds.

I've tested **linux-5.4.2.arch1-1** and **linux-5.4.8.arch1-1** kernels, both have the same problem, after fallback to **5.3.13-arch1-1** evdi working as it should.

**Comment by asterlab** [comment]
got the same issue with kernel 5.4.0-2-amd64

**Comment by rghv** [comment]
Can validate; witnessed the same on 5.4.12 (taken via ukuu) on Linux Mint 19.3 Tricia.

Noticed issues similar to https://github.com/DisplayLink/evdi/issues/180 and https://github.com/DisplayLink/evdi/issues/160 as well.

**Comment by abucodonosor** [comment]
Are you guys using Intel GPU(s)?

**Comment by kvaps** [comment]
> Are you guys using Intel GPU(s)?

My system:

* **OS:** Arch Linux x86_64 
* **Host:** Lenovo ideapad 720S-13ARR 
* **Kernel:** 5.3.10-arch1-1 
* **CPU:** AMD Ryzen 7 2700U
* **GPU:** AMD Radeon RX Vega 10

**Comment by xtrasimplicity** [comment]
> Are you guys using Intel GPU(s)?

I'm having this problem on a Dell XPS 13 7390 2-in-1 (2019 model), with an Intel Iris GPU.

Evdi was rather verbose with its logging, but I can't remember what it said (and I'm not near my dock). Will post the log when I'm back at my desk.

**Comment by abucodonosor** [comment]
@kvaps 

can you attach a dmesg from a broken kernel?

You can boot such a kernel, let it break then reboot to a good one and use journalctl
to show the dmesg. Maybe you already have such a dmesg, just use journalctl --list-boots to see your entries then journalctl -b <the-entrie> -o cat >dmesg.txt

**Comment by abucodonosor** [comment]
Also for people with Intel HW and this issue please see https://github.com/DisplayLink/evdi/issues/173

**Comment by kvaps** [comment]
@abucodonosor I'll do this, short note: kernel is not hanging fully, its got unstuck after certain amount of time

**Comment by abucodonosor** [comment]
@kvaps 

even better, for sure there are some traces in the dmesg then.

**Comment by kvaps** [comment]
Hi @abucodonosor, after every  stuckage I see the following messages:
```
[  133.132542] [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx timeout, but soft recovered
[  161.293215] [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx timeout, but soft recovered
[  190.946188] [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx timeout, but soft recovered
[  200.973338] [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx timeout, but soft recovered
[  214.199950] [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx timeout, but soft recovered
```

**Comment by kvaps** [comment]
Problem solved for me by adding `iommu=pt` to my kernel cmdline

**Comment by abucodonosor** [comment]
@kvaps 

that is a kernel bug itself it seems. Adding iommu=pt is a workaround, maybe it is fixed in 5.5.

**Comment by kvaps** [comment]
OK, let's see.

**Comment by kvaps** [comment]
@abucodonosor, you're right it seems no more freezes on 5.5 even without `iommu=pt`

**Comment by abucodonosor** [comment]
@kvaps 

Good. They fixed a similar bug on Intel GPUs in 5.5 also.
